Test/Resolve-WhiskeyTaskPathParameter.Tests.ps1

```diff
@@ -0,0 +1,692 @@
+        }
+        Init
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = 'abc.yml' } -ErrorAction SilentlyContinue
+        ThenTaskNotCalled
+        ThenThrewException -Pattern 'does\ not\ exist'
+    }
+}
+
+Describe ('Resolve-WhiskeyTaskPathParameter.when path is absolute') {
+    It ('should succeed') {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath(Mandatory,MustExist=$false,PathType='File')]
+                [string]$Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = [IO.Path]::GetFullPath((Join-Path -Path $TestDrive.FullName -Childpath 'abc.yml' )) }
+        ThenTaskCalled -WithParameter @{ 'Path' = [IO.Path]::GetFullPath((Join-Path -Path $TestDrive.FullName -Childpath 'abc.yml' )) }
+        ThenPipelineSucceeded
+    }
+}
+
+Describe ('Resolve-WhiskeyTaskPathParameter.when path doesn''t exist but has a wildcard') {
+    It ('should work as expected') {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath(Mandatory,MustExist=$false,PathType='File')]
+                [string]$Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        WhenRunningTask 'Task' -Parameter @{ 'Path' = 'packages\tool\*.yolo' } -ErrorAction SilentlyContinue
+        if ($PSVersionTable.PSVersion.Major -lt 6)
+        {            
+            ThenTaskNotCalled
+            ThenThrewException -Pattern 'Illegal\ characters\ in\ path.'
+        }
+        else
+        {
+            ThenTaskCalled -WithParameter @{ 'Path' = (Join-Path -Path $TestDrive.FullName -ChildPath 'packages\tool\*.yolo') }
+            ThenPipelineSucceeded
+        }
+    }
+}
+
+Describe ('Resolve-WhiskeyTaskPathParameter.when path does not start with the correct BuildRoot') {
+    It ('should fail') {
+        function global:Task
+        {
+            [Whiskey.Task('Task')]
+            param(
+                [Whiskey.Tasks.ValidatePath(Mandatory,MustExist=$false,PathType='File')]
+                [string]$Path
+            )
+            $global:taskCalled = $true
+            $global:taskParameters = $PSBoundParameters
+        }
+        Init
+        WhenRunningTask -Name 'Task' `
+                        -Parameter @{ 'Path' = ('..\' + (Split-Path -Path $TestDrive.FullName -Leaf) + '!\abc.yml') } `
+                        -BuildRoot ($TestDrive.FullName + '///') `
+                        -ErrorAction SilentlyContinue
+        ThenTaskNotCalled 
+        ThenThrewException -Pattern 'outside\ the\ build\ root'
+    }
+}
+
+Describe ('Resolve-WhiskeyTaskPathParameter.when path contains wildcards and "../"') {

```

Comment:
I don't think you need this test? It's testing two states covered by other tests?